Integrations/build.gradle

```diff
@@ -67,7 +67,7 @@ def runInDocker = { String name, String sourcePath, List<String> command, Closur
             // set up the container, env vars - things that aren't likely to change
             from 'deephaven/server-netty:local-build'
             runCommand '''set -eux; \\
-                      pip3 install unittest-xml-reporting==3.0.4; \\
+                      pip3 install unittest-xml-reporting==3.0.4 connectorx; \\
```

Comment:
connectorx should at a bare minimum be listed as an optional dependency.  I don't see that happening anywhere in the PR.
https://setuptools.pypa.io/en/latest/userguide/dependency_management.html

Reply:
I believe this is one of the packages (including turbodbc, adbc) we deemed as 'need to be installed by the user' in his/her Python environment. We could add it to the optional required packages for deephaven-core the same as we did for 'jedi', but then we need to include turbodbc, adbc as well. The pip command syntax for installing addition features of a package isn't the most friendly to remember. I think we might be better off either making it a default requirement or leaving it out entirely. @devinrsmith , your thoughts too?

Reply:
Our "Integration" testing image includes packages we don't explicitly mark as optional - see https://github.com/deephaven/deephaven-server-docker/blob/main/contexts/server-base/requirements.txt.

(It probably makes sense to move `unittest-xml-reporting` to there as well.)

That said, we _could_ add optional features in our setup.py - I'm not sure the best way to organize them, if dbc is separate from adbc is separate from odbc?

An alternative that I don't think we've considered is to have "proper" additional python packages:

`deephaven-core-odbc`, `deephaven-core-autocomplete`, etc. In this way, we _could_ create optional packages that have concrete requirements.